### DISCUSSION TITLE
Is it just me, or is anyone else crashing after the first epoch?

### DISCUSSION THREAD

**Discussion by ye7iaserag** [d8ahazard/sd_dreambooth_extension#932] [Q&A]
I'm using a 3080ti and I have been training 1 epoch at a time for the past 2 weeks since the extension is crashing every epoch and I have to restart the whole webui to resume training.
The problem is after the first epoch a checkpoint is generated and a sample then the memory climbs up and crashes due to out of memory, even if I disable both checkpoint and sample generation the problem happens.
I've created an issue https://github.com/d8ahazard/sd_dreambooth_extension/issues/835 but not sure if it's happening with anyone else so...
Anyone else facing this?

**Comment by morphinapg** [comment]
Having the same issue trying to train 2.1 768. I figured I just don't have enough VRAM to train that model on my GPU perhaps. Maybe someone might have some suggestions to reduce VRAM, but I think I might have tried everything.

**Reply by ye7iaserag** [reply]
For me it happenes with 1.5, previously I was able to train 2.1 but without the text encoder and without generating samples.
So there must be a new change that causes vram usage to increase between epochs or some memory that doesn't get cleaned.
I think overall the extension might need some optimizations in regards to vram management, to achive maximum efficiency, load only what is needed at a specific step and nothing else, I know reloading stuff in and out of vram will affect performance, but we can have different strategies for vram management, optimized vs performance modes.

**Comment by Arilziem** [comment]
I believe there have been some improvements in the dev branch that slightly reduce VRAM usage, but I think you might just have too little VRAM to train UNET and text encoder at the same time. Try setting the text encoder slider to 0 in the advanced settings. You can train the text encoder separately by unchecking 'Train UNET'.

Another thing you could try is switching attention to flash_attention, that might reduce VRAM usage just enough to train both. Note that you need to create a new model when switching attention type.

**Reply by morphinapg** [reply]
I actually HAD to disable the text encoder to even be able to train the first epoch. Before that, I was just barely short of VRAM to even start, but it still ran out of VRAM after the first epoch. Personally, I didn't even get the sample to render. I have the same GPU, 3080Ti

**Reply by Arilziem** [reply]
I have been on the dev branch for a while now but I am pretty sure I switched because of this very issue with generating samples, which works flawlessly now for my non-TI 3080 on dev. You could try disabling samples if that is an option for you.

**Reply by ye7iaserag** [reply]
But was this something that changed in the past couple of months? Because I was training 1.5 models before and I used to run 100k steps over a lot of epochs without any problems

**Reply by Arilziem** [reply]
Yes, there were some issues introduced in December after restructuring iirc. VRAM management works fine in dev, so hopefully there will a stable version soon that can be merged back into main.
